src/components/facets/FacetConnected.tsx

```diff
@@ -20,7 +20,7 @@ const mapStateToProps = (state: IReactVaporState, ownProps: IFacetOwnProps): IFa
 
 const getSelectedRows = (rows: IFacet[], selected: IFacet[]): IFacet[] => _.without(_.map(rows, (row: IFacet) => {
     const element = _.findWhere(selected, {name: row.name});
-    return element ? _.extend(element, row) : null;
+    return element ? _.extend({}, element, row) : null;

```

Comment:
I think `return element && {...element, ...row};` would do the same and is shorter.